torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -186,4 +185,134 @@ torch::CppFunction autogradNotImplementedFallback() {
+  // Mimics a subset of the logic from ADInplaceOrViewType kernel
+  // - see gen_inplace_or_view_type.py
+  // - this should only be used with autogradNotImplementedFallback above
+  //
+  // NOTE [ Limitations of ADInplaceOrView boxed kernel ]
+  //
+  // This op should only be used with autogradNotImplementedFallback kernel
+  // because there is some logic we need specifically to enforce that even
+  // if we do in-place on view's created in this kernel, the proper "derivative
+  // is not implemented" error is still raised.
+  //
+  // For views:
+  // - We enforce that the view relationship is between the first input
+  //   and the first output (which may be either Tensor or vec of Tensor)
+  //
+  // For inplace ops:
+  // - (TODO?) enforce that the same op cannot be both a view and
+  //   that is not allowed in the gen_inplace_or_view logic
+  //
+  // TODO: We could just link to the documentation once we have that
+  // instead of repeating this.
+  const auto& schema = op.schema();
+  const auto& op_name = schema.operator_name().name;
+  const auto& arguments = schema.arguments();
+  const auto& returns = schema.returns();
+  const auto num_arguments = arguments.size();
+  const auto num_returns = returns.size();
+  const auto stack_start = stack->size() - num_arguments;
+  bool any_is_inplace = false;
+
+  at::Tensor aliased_input;
+
+  int64_t aliased_output_idx = -1;
+  for (const auto i : c10::irange(num_returns)) {
+    const auto& alias_info = returns[i].alias_info();
+    if (alias_info.has_value() && !alias_info->isWrite()) {
+      TORCH_CHECK(
+        aliased_output_idx == -1,
+        "Fallback ADInplaceOrView kernel expects only a single output in the operator schema to have a "
+        "non-write alias annotation (i.e., 'Tensor(a)'). "
+        "Non-composite functions where multiple outputs are aliased with inputs aren't supported."
+        "Please rewrite your function as a composite function.");
+      aliased_output_idx = i;
+    }
+  }
+
+  int64_t aliased_input_idx = -1;
+  for (const auto i : c10::irange(num_arguments)) {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value()) {
+      if (!alias_info->isWrite()) {
+        TORCH_CHECK(
+          aliased_input_idx == -1,
+          "Fallback ADInplaceOrView kernel expects only a single input in the operator schema to have a "
+          "non-write alias annotation (i.e., 'Tensor(a)'). "
+          "Non-composite functions where multiple inputs are aliased with outputs aren't supported. "
+          "Please rewrite your function as a composite function.");
+        aliased_input_idx = i;
+        const c10::IValue& aliased_input_iv = (*stack)[stack_start + i]; // get a reference to an ivalue on the stack
+        TORCH_CHECK(aliased_input_iv.isTensor());
+        // copy assignment (TODO: how to avoid this extra reference count bump??)
+        // It doesn't seem like we can save merely a pointer to Tensor because it would
```

Comment:
Why would it get dropped from the stack? During the redispatch below?
I think this extra call is ok tbh. Maybe clean up the comment a bit?

Reply:
Yeah, I think it should get dropped from the stack during redispatch. Will just clean up the comment for now.